chainerrl/links/noisy_chain.py

```diff
@@ -50,5 +50,14 @@ def _map_links(func, link):
                 children[i].name = str(i)
 
                 if isinstance(link, Sequence):
-                    # assumes i-th layer corresponds with i-th child
-                    link.layers[i] = new_child
+                    _replace_unique_item(link.layers, child, new_child)
+                # Check chainer.Sequential if it exists.
+                sequential_class = getattr(chainer, 'Sequential', ())
+                if isinstance(link, sequential_class):
```

Comment:
Why not say `if isinstance(link, chainer.Sequential):`?

Reply:
To be consistent with `isinstance(link, chainer.ChainList)`.

Reply:
> Why not say `if isinstance(link, chainer.Sequential):`?

Because we've not dropped the Chainer v4 support.  The code is almost equivalent with
```python
if hasattr(chainer, 'Sequential') and isinstance(link, chainer.Sequential):
```
but `hasattr` was avoided.